src/models/maze/initialState.ts

```diff
@@ -1,65 +1,39 @@
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.empty, visited: false, path: false }
-    ],
-    2: [
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false }
-    ],
-    3: [
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.end, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false }
-    ]
-  },
-  clearMaze: {
-    0: [
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.start, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false }
-    ],
-    1: [
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false }
-    ],
-    2: [
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false }
-    ],
-    3: [
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.end, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false },
-      { type: SpaceTypes.wall, visited: false, path: false }
-    ]
+export const generateMaze = (
+  x: number,
+  y: number,
+  clear?: boolean
+): MazeInfo => {
+  let maze: MazeInfo = {};
+  for (let i = 0; i < x; i++) {
+    maze[i] = [];
+    for (let j = 0; j < y; j++) {
+      if (i === 0 && j === 0 && !clear)
+        maze[i].push({
+          type: SpaceTypes.start,
+          visited: false,
+          path: false
+        });
+      else if (i === x - 1 && j === y - 1 && !clear)
+        maze[i].push({
+          type: SpaceTypes.end,
+          visited: false,
+          path: false
+        });
+      else
+        maze[i].push({
+          type: SpaceTypes.wall,
+          visited: false,
+          path: false
+        });
+    }
   }
+  return maze;
+};
+
+export const initialState: Maze = {
+  mazeInfo: generateMaze(5, 5),
+  clearMaze: generateMaze(5, 5, true)
```

Comment:
Do we still need clearMaze since the reducer just uses your generateMaze function?